src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -53,33 +55,48 @@ public override void Initialize (AnalysisContext context)
 					}
 				}
 			});
-			// TODO: fix reporting for generic type substitutions. This should happen not only for method invocations,
-			// but for any reference to an instantiated method or type.
-			context.RegisterOperationAction (context => {
-				var invocationOperation = (IInvocationOperation) context.Operation;
-				ProcessInvocationOperation (context, invocationOperation);
-			}, OperationKind.Invocation);
+			context.RegisterSyntaxNodeAction (context => {
+				ProcessGenericParameters (context);
+			}, SyntaxKind.GenericName);
 		}
 
-		static void ProcessInvocationOperation (OperationAnalysisContext context, IInvocationOperation invocationOperation)
+		static void ProcessGenericParameters (SyntaxNodeAnalysisContext context)
 		{
-			if (context.ContainingSymbol.IsInRequiresUnreferencedCodeAttributeScope ())
+			// We dont need to check for RUC for INamedTypeSymbol here
```

Comment:
I think this warrants a more detailed comment - maybe something along these lines:

RUC on the containing symbol normally silences warnings, but when examining a generic base type, the containing symbol is the declared derived type. RUC on the derived type does not silence warnings about base type arguments.